lib/cext/preprocess.rb

```diff
@@ -11,6 +11,26 @@
 
 ALLOCA_LOCALS = /^(\s+)VALUE\s*\*\s*([a-z_][a-zA-Z_0-9]*)\s*=\s*(\(\s*VALUE\s*\*\s*\)\s*)?alloca\(/
 
+# Found in nokogiri
+XML_NODE_SET_PATCH = <<-EOF
+  switch (rb_tr_to_int_const(rb_range_beg_len(arg, &beg, &len, (long)node_set->nodeNr, 0))) {
+  case Qfalse_int_const:
+    break;
+  case Qnil_int_const:
+EOF
+
+# Found in pg
+PG_BINARY_ENCODER_PATCH = <<-EOF	
+	switch(rb_tr_to_int_const(value)){
+		case Qtrue_int_const : mybool = 1; break;
+		case Qfalse_int_const : mybool = 0; break;
+EOF
+
+PATCHED_FILES = {'xml_node_set.c'      => {:line =>  "  switch (rb_range_beg_len(arg, &beg, &len, (long)node_set->nodeNr, 0)) {\n", 
+                                           :patch => XML_NODE_SET_PATCH},
+                 'pg_binary_encoder.c' => {:line => "	switch(value){\n",
+                                           :patch => PG_BINARY_ENCODER_PATCH}}
```

Comment:
I think it's generally useful to replace repeating pattern like `foo->bar` when we know we need to de-handle that. Line number break immediately with any added or removed line.
In this case I would match what happen after that switch, and look for `case Qfalse:` as well. Maybe this can be generalized easily to a switch using `Q` cases.